glean-core/src/upload/mod.rs

```diff
@@ -69,6 +77,74 @@ pub fn setup_upload_manager(upload_manager: PingUploadManager) -> Result<()> {
     Ok(())
 }
 
+/// A counter to help limit the number of processing pings per minute.
+///
+/// NOTE: A "processing" ping is a ping that has been dequeued through `get_upload_task`.
+#[derive(Debug)]
+struct PingCounter {
+    started: Option<SystemTime>,
+    count: u32,
+}
+
+impl PingCounter {
+    pub fn new() -> Self {
+        Self {
+            started: None,
+            count: 0,
+        }
+    }
+
+    fn reset(&mut self) {
+        self.started = Some(SystemTime::now());
```

Comment:
Is this clock monotonic? We should use a monotonic clock  :) We have "chrono" for that, too.